### DISCUSSION TITLE
JDT.LS Not Recognizing Project Structure in Neovim Unless Imported into Eclipse

### DISCUSSION THREAD

**Discussion by kojogadget** [mfussenegger/nvim-jdtls#547] [General]
I don't know if this is a nvim-jdtls issue, or a eclipse.jdt.ls issue, or just a issue as a user.. But I have posted the issue at eclipse.jdt.ls. Also posting it here to see if any has a simple solution. 

Issue: https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues/2816

I see some others have experienced the same, but their solution didn't solve it in my case.

**Comment by jaadth** [comment]
I had some problems at first using this wonderful plugin with a project am working on.  (wonderful in that it has allowed me to keep using vi as my ide)  😀 


Directory Layout:

repo/.git
repo/svc1/build.gradle
repo/svc2/build.gradle
repo/svc3/build.gradle
repo/master/gradlew

I think this is referred to as a multi-project layout.

So I had to make a custom find_root() function that traverses up the tree and always looks into each subdirectory to find the project root.

I am a vi/vim/nvim C Makefile developer for 20+ years so I really just didn't know if this was just a really unique project or if others had the same problems.

I do know that others in my organization who have been career JAVA developers also had problems sorting out their IDE's.

Well, hope this helps someone out.
